Source/WebCore/dom/MessagePort.cpp

```diff
@@ -108,6 +128,9 @@ MessagePort::MessagePort(ScriptExecutionContext& scriptExecutionContext, const M
 
     Locker locker { allMessagePortsLock };
     allMessagePorts().set(m_identifier, this);
+    ensureOnMainThread([portIdentifier = m_identifier, scriptExecutionContextIdentifier = scriptExecutionContext.identifier()] {
```

Comment:
What happens if code above runs before this gets to execute in the main thread?
Maybe it's better to have a lock for this map.

Reply:
Then ensureOnMainThread() will run its lambda synchronously. I don't understand the problem?

Reply:
Even if we were currently not on the main thread? Wouldn't it schedule a function call in the main thread in that case?

Reply:
I believe this is safe, even in the async case (where we need to dispatch to the main thread).
That said, we can certainly use a lock and do it straight from the background thread. I was trying to reduce locking but it may not be a worthy optimization here.